src/modules/data/shared/move/sagas.js

```diff
@@ -0,0 +1,211 @@
+			headers: {
+				'Content-Type': 'application/x-www-form-urlencoded;charset=UTF-8',
+			},
+			credentials: 'include',
+		} );
+
+		return yield call( [ response, 'json' ] );
+	} catch ( error ) {
+
+	}
+}
+
+/**
+ * Fetches usable oost types
+ * @returns {Object} JSON response
+ */
+export function* fetchPostTypes() {
+	try {
+		yield put( {
+			type: types.FETCH_POST_TYPES,
+		} );
+		const { data } = yield call( _fetch, {
+			action: 'move_tickets_get_post_types',
+		} );
+		yield put( {
+			type: types.FETCH_POST_TYPES_SUCCESS,
+			data,
+		} );
+		return data;
+	} catch ( error ) {
+		yield put( {
+			type: types.FETCH_POST_TYPES_ERROR,
+			error,
+		} );
+	}
+}
+
+/**
+ * Fetches filtered posts based on criteria
+ *
+ * @export
+ * @param {*} {
+ * 	ignore,
+ * 	post_type,
+ * 	search_terms = '',
+ * }
+ * @returns {Object} JSON response
+ */
+export function* fetchPostChoices( {
+	ignore,
+	post_type,
+	search_terms = '',
+} ) {
+	try {
+		yield put( {
+			type: types.FETCH_POST_CHOICES,
+		} );
+		const { data } = yield call( _fetch, {
+			action: 'move_tickets_get_post_choices',
+			ignore,
+			post_type,
+			search_terms,
+		} );
+		yield put( {
+			type: types.FETCH_POST_CHOICES_SUCCESS,
+			data,
+		} );
+		return data;
+	} catch ( error ) {
+		yield put( {
+			type: types.FETCH_POST_CHOICES_ERROR,
+			error,
+		} );
+	}
+}
+
+/**
+ * Moves ticket/RSVP from one post to another
+ *
+ * @export
+ * @param {*} {
+ * 	src_post_id,
+ * 	ticket_type_id,
+ * 	target_post_id,
+ * }
+ * @returns {Object} JSON response
+ */
+export function* moveTicket( {
+	src_post_id,
+	ticket_type_id,
+	target_post_id,
+} ) {
+	try {
+		yield put( {
+			type: types.MOVE_TICKET,
+		} );
+		const { data } = yield call( _fetch, {
+			action: 'move_ticket_type',
+			src_post_id,
+			ticket_type_id,
+			target_post_id,
+		} );
+		yield put( {
+			type: types.MOVE_TICKET_SUCCESS,
+			data,
+		} );
+		return data;
+	} catch ( error ) {
+		yield put( {
+			type: types.MOVE_TICKET_ERROR,
+			error,
+		} );
+	}
+}
+
+export function* getCurrentPostId() {
+	return yield call( [ wpSelect( 'core/editor' ), 'getCurrentPostId' ] );
+}
+
+export function* getPostChoices() {
+	const params = yield all( {
+		post_type: select( selectors.getModalPostType ),
+		search_terms: select( selectors.getModalSearch ),
+		ignore: call( getCurrentPostId ),
+	} );
+	yield call( fetchPostChoices, params );
+}
+
+export function* onModalChange( action ) {
+	if (
+		! action.payload.hasOwnProperty( 'target_post_id' ) &&
+		! action.payload.hasOwnProperty( 'ticketId' )
+	) {
+		yield call( delay, 500 );
+		yield call( getPostChoices );
+	}
+}
+
+export function* onModalSubmit() {
+	const params = yield all( {
+		src_post_id: call( getCurrentPostId ),
+		target_post_id: select( selectors.getModalTarget ),
+		ticket_type_id: select( selectors.getModalTicketId ),
+	} );
+	yield fork( moveTicket, params );
+
+	const action = yield take( [ types.MOVE_TICKET_SUCCESS, types.MOVE_TICKET_ERROR ] );
```

Comment:
It'll still proceed through, as it will take either the success or error action. No second take in the case, the user would have to submit the form again to go through this again. On the error action nothing would happen